appserver/payara-appserver-modules/microprofile/config-extensions/src/main/java/fish/payara/microprofile/config/extensions/hashicorp/HashiCorpSecretsConfigSource.java

```diff
@@ -0,0 +1,233 @@
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package fish.payara.microprofile.config.extensions.hashicorp;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+import java.io.StringReader;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.json.Json;
+import javax.json.JsonException;
+import javax.json.stream.JsonParser;
+import javax.json.stream.JsonParser.Event;
+import javax.ws.rs.ProcessingException;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.glassfish.config.support.TranslatedConfigView;
+import org.jvnet.hk2.annotations.Service;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import fish.payara.microprofile.config.extensions.hashicorp.model.SecretHolder;
+import fish.payara.nucleus.microprofile.config.source.extension.ConfiguredExtensionConfigSource;
+
+@Service(name = "hashicorp-secrets-config-source")
+public class HashiCorpSecretsConfigSource extends ConfiguredExtensionConfigSource<HashiCorpSecretsConfigSourceConfiguration> {
+
+    private static final Logger LOGGER = Logger.getLogger(HashiCorpSecretsConfigSource.class.getName());
+
+    private Client client = ClientBuilder.newClient();
+    protected String hashiCorpVaultToken;
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    @Override
+    public void bootstrap() {
+        try {
+            // Get the HashiCorp Vault token.
+            hashiCorpVaultToken = TranslatedConfigView.getRealPasswordFromAlias("${ALIAS=HASHICORP_VAULT_TOKEN}");
+        } catch (KeyStoreException | CertificateException | IOException | NoSuchAlgorithmException | UnrecoverableKeyException ex) {
+            LOGGER.log(Level.WARNING, "Unable to get value from password aliases", ex);
+
+        }
+    }
+
+    @Override
+    public Map<String, String> getProperties() {
+        Map<String, String> results = new HashMap<>();
+
+        if (hashiCorpVaultToken == null) {
+            printMisconfigurationMessage();
+            return results;
+        }
+
+        final WebTarget secretsTarget = client
+                .target(configuration.getVaultAddress() + "/v1" + configuration.getPath());
```

Comment:
This line is confusing as a user. Should the path include the `/data` for a v2 API? Should it include the `/kv`? If this feature only supports KV stores I'd like to see that taken. I'd like this cleared up and explained in the admin console. Also I'd like to see failed responses log the endpoint that is actually called so they can debug it